### PR TITLE
add CMake3 support

### PR DESCRIPTION
**Pull request by kostyfisik**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,62 +57,120 @@ elseif (${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
...
-  install (TARGETS ${PROJECT_NAME}_static DESTINATION lib)
+include(GNUInstallDirs)
+SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
+SET(CMAKE_Fortran_MODULE_DIRECTORY "${PROJECT_BINARY_DIR}/modules")
+
+###############################################################################
+set(LIB complex_bessel)
+add_library(${LIB}
```
Comment (joeydumont) [inline CMakeLists.txt:68]: I would prefer to build a shared library by default. Just adding `SHARED` will do, unless you want to set up an optionally `BUILD_STATIC` variable, like I had started doing but never finished...

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,39 +3,31 @@
...
+# The target of the complex_bessel library is exported
+# as complex_bessel::complex_bessel to a package configuration file for this
+# library
+#
+# usage:
+#     find_package(complex_bessel) or add_subdirectory(complex_bessel)
+# and then for your <target>:
+#     target_link_library(<target> complex_bessel::complex_bessel)
+#
+# the config file is generatet in the build and install directories
```
Comment (joeydumont) [inline CMakeLists.txt:21]: typo: "generated"

File: CMakeLists.txt [hunk 3]
```diff
@@ -65,62 +57,120 @@ elseif (${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
...
+
+
+# testing
+if(${PROJECT_SOURCE_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
+  set(main_project TRUE)
+  option(BUILD_TESTING "Build the testing tree." OFF)
+else()
+  set(main_project FALSE)
+
+  # if this is not the main project but BUILD_TESTIG is set to TRUE
```
Comment (joeydumont) [inline CMakeLists.txt:91]: typo: "BUILD_TESTING"

File: README.md [hunk 4]
```diff
@@ -43,7 +43,30 @@ it, you will have to run `cmake` manually like so:
...
+The target of the complex_bessel library is exported
+as `complex_bessel::complex_bessel` to a package configuration file for this
+library.
+
+After installation you can find this library in your
+project CMakeLists.txt with:
+  ```cmake
+  find_package(complex_bessel)
+```
+or you can just put a copy of `complex_bessel` source code with zero changes
```
Comment (joeydumont) [inline README.md:56]: Remove "zero changes". They can just put a copy of the source.

File: README.md [hunk 5]
```diff
@@ -43,7 +43,30 @@ it, you will have to run `cmake` manually like so:
...
+into your source tree and just add it from the upper level CMakeLists.txt
+  ```cmake
+ add_subdirectory(complex_bessel)
+```
+To compile and link this library you should have C\C++14 and Fortran compiles
```
Comment (joeydumont) [inline README.md:61]: typo: C++14 and Fortran compilers*

**Review by joeydumont — CHANGES_REQUESTED** [review]
I haven't tested all changes,and I haven't looked into how the different COMPONENT directives work, but this looks good, apart from a few typos and building a static library intead of a shared one by default. I was able to compile and run tests on my machine, and CI also completed fine. 

I'll approve when the minor changes are completed! Thanks!

**Comment by kostyfisik** [comment]
OK, I will work on this next week

File: CMakeLists.txt (hunk 2 above)
Comment (kostyfisik) [reply CMakeLists.txt:21]: I know about this typo, if you don't mind I would keep it. The reason is that the initial example CMakeList.txt has this typo, so it is kind of a fingerprint to show this origin. I know a dozen of other CMake projects wich keep this typo.

File: README.md (hunk 5 above)
Comment (kostyfisik) [reply README.md:61]: It seems that you have to use pure C for linking (if I remove C from cmake project description the linking fails)

File: CMakeLists.txt (hunk 1 above)
Comment (kostyfisik) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 3 above)
Comment (kostyfisik) [reply CMakeLists.txt]: done

File: README.md (hunk 4 above)
Comment (kostyfisik) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (kostyfisik) [reply README.md]: Sorry, C was only for linkage with hdf5. Updated.

File: CMakeLists.txt (hunk 2 above)
Comment (joeydumont) [reply CMakeLists.txt:21]: You've already put the link in the header. I'd rather have the typo fixed.

Comment (kostyfisik) [reply CMakeLists.txt]: done

**Comment by joeydumont** [comment]
Thank you for your contribution!

**Comment by kostyfisik** [comment]
@joeydumont Thank you for a very usefull lib! Amos implementation of Bessel functions for complex argument is the best one among open source, I was really suprised that it is not available in any major C++ lib.

**Pull request merged**